### ISSUE TITLE
".mono/registry" files in git repo

### ISSUE THREAD

**Issue by JoeGaggler** [microsoft/vscode-dotnettools#981]
Type: <b>Bug</b>

In the latest version, this extension is adding files to all repos, in the ".mono/registry" folder. I have added this to ".gitignore", but this was not necessary before. The extension should store internal values somewhere else.

Extension version: 1.4.28
VS Code version: Code 1.87.1 (1e790d77f81672c49be070e04474901747115651, 2024-03-06T00:22:39.156Z)
OS version: Darwin arm64 23.4.0
Modes:

**Comment by JoeGaggler** [comment]
> Are you repos in your user/home directory?
Yes, my repos are in a path like `/Users/joegaggler/code/myrepohere`

> Could you share the name (and if possible contents) of one of the files at the end of the folders to help us understand the issue?

One recent file (masked slightly), `.mono/registry/CurrentUser/software/microsoft/csdevkit/v1/pids/123_456789012345678901/values.xml`:
```xml
<values>
  <value name="StillAlive" type="qword">123456789012345678</value>
</values>
```

**Comment by jonathanjyi** [comment]
thanks @JoeGaggler, is there a `.mono/registry/CurrentUser/software/microsoft/csdevkit/telemetry` path as well, or just the `v1`?

**Comment by JoeGaggler** [comment]
I have been aggressively purging those files, but the paths are still present, yes:
<img width="610" alt="image" src="https://github.com/microsoft/vscode-dotnettools/assets/27233871/26a4c9e6-c34b-4548-aea8-7e71300662ca">

I do not see `HOMEDIR` set.

**Comment by jonathanjyi** [comment]
@JoeGaggler sorry, not HOMEDIR, `HOME`. 
`echo $HOME`

**Comment by JoeGaggler** [comment]
Ah, yes, `HOME` is set to `/Users/joegaggler`.

**Comment by jonathanjyi** [comment]
thanks @mikeharder and @JoeGaggler, we'll be investigating why this is occuring.

**Comment by JoeGaggler** [comment]
Can you direct me to information on how to turn off all telemetry?

**Comment by jonathanjyi** [comment]
Hi Joe, you can turn off all telemetry following the instructions here: https://code.visualstudio.com/docs/getstarted/telemetry#_disable-telemetry-reporting

Please keep in mind that although this will disable sending of telemetry, it will not disable the creation of the files that you have been seeing as those are used by C# Dev Kit for internal services.

**Comment by mauve** [comment]
I get these files as well (WSL2)

.mono/registry/CurrentUser/software/microsoft/csdevkit/telemetry/persistentpropertybag/microsoft.visualstudio.code.servicehost/values.xml

**Comment by abaines** [comment]
I'm using WSL (VSC) and getting these:
![collection of value.xml files](https://github.com/microsoft/vscode-dotnettools/assets/4323841/9a442db6-cd1b-4de0-98c8-dad3b7d24aaf)
It is in a project I spun up today so I can't say if it was happening before.

If I delete the `.mono` folder and in VSC use the `Reload Window`, they seem to come back.

**Comment by jonathanjyi** [comment]
Duplicate of #1008

**Comment by jonathanjyi** [comment]
Duplicate of $1008

**Comment by JoeGaggler** [comment]
Setting the `.mono` folder with read-only permissions was super effective ;-)
